qadence/analog/interaction.py

```diff
@@ -139,56 +143,94 @@ def _(
+    b: AbstractBlock,
+    r: Register,
+    interaction: Callable,
+    pattern: Union[AddressingPattern, None],
+) -> AbstractBlock:
     raise NotImplementedError(f"Cannot emulate {type(b)}")
 
 
 @_add_interaction.register
-def _(b: CompositeBlock, r: Register, i: Callable) -> AbstractBlock:
-    return _construct(type(b), tuple(map(lambda b: _add_interaction(b, r, i), b.blocks)))
+def _(
+    b: CompositeBlock,
+    r: Register,
+    i: Callable,
+    pattern: Union[AddressingPattern, None],
+) -> AbstractBlock:
+    return _construct(type(b), tuple(map(lambda b: _add_interaction(b, r, i, pattern), b.blocks)))
 
 
 @_add_interaction.register
-def _(block: ScaleBlock, register: Register, interaction: Callable) -> AbstractBlock:
+def _(
+    block: ScaleBlock,
+    register: Register,
+    interaction: Callable,
+    pattern: Union[AddressingPattern, None],
+) -> AbstractBlock:
     if isinstance(block.block, AnalogBlock):
         raise NotImplementedError("Scaling emulated analog blocks is not implemented.")
     return block
 
 
 @_add_interaction.register
-def _(block: PrimitiveBlock, register: Register, interaction: Callable) -> AbstractBlock:
+def _(
+    block: PrimitiveBlock,
+    register: Register,
+    interaction: Callable,
+    pattern: Union[AddressingPattern, None],
+) -> AbstractBlock:
     return block
 
 
 @_add_interaction.register
-def _(block: WaitBlock, register: Register, interaction: Callable) -> AbstractBlock:
+def _(
+    block: WaitBlock,
+    register: Register,
+    interaction: Callable,
+    pattern: Union[AddressingPattern, None],
+) -> AbstractBlock:
     duration = block.parameters.duration
 
     support = tuple(range(register.n_qubits))
     assert support == block.qubit_support if not block.qubit_support.is_global else True
     pairs = list(filter(lambda x: x[0] < x[1], product(support, support)))
 
-    return HamEvo(interaction(register, pairs), duration / 1000) if len(pairs) else I(0)
+    p_terms = add_pattern(register, pattern)
+    generator = interaction(register, pairs) + p_terms
+
+    return HamEvo(generator, duration / 1000) if len(pairs) else I(0)
 
 
 @_add_interaction.register
-def _(block: ConstantAnalogRotation, register: Register, interaction: Callable) -> AbstractBlock:
+def _(
+    block: ConstantAnalogRotation,
+    register: Register,
+    interaction: Callable,
+    pattern: Union[AddressingPattern, None],
+) -> AbstractBlock:
     # convert "global" to indexed qubit suppport so that we can re-use `kron` dispatched function
     b = deepcopy(block)
     b.qubit_support = QubitSupport(*range(register.n_qubits))
-    return _add_interaction(kron(b), register, interaction)
+    return _add_interaction(kron(b), register, interaction, pattern)
 
 
 @_add_interaction.register
-def _(block: AnalogKron, register: Register, interaction: Callable) -> AbstractBlock:
+def _(
+    block: AnalogKron,
+    register: Register,
+    interaction: Callable,
+    pattern: Union[AddressingPattern, None],
```

Comment:
Why not make `pattern` as a keyworded argument?

Reply:
This will be changed anyway in @jpmoutinho's PR.